stackit/internal/services/serverupdate/schedule/resource.go

```diff
@@ -0,0 +1,452 @@
+		return
+	}
+	tflog.Info(ctx, "Server update schedule deleted.")
+}
+
+// ImportState imports a resource into the Terraform state on success.
+// The expected format of the resource import identifier is: // project_id,server_id,schedule_id
+func (r *scheduleResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	idParts := strings.Split(req.ID, core.Separator)
+	if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing server update schedule",
+			fmt.Sprintf("Expected import identifier with format [project_id],[server_id],[update_schedule_id], got %q", req.ID),
+		)
+		return
+	}
+
+	intId, err := strconv.ParseInt(idParts[2], 10, 64)
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics,
+			"Error importing server update schedule",
+			fmt.Sprintf("Expected update_schedule_id to be int64, got %q", idParts[2]),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("project_id"), idParts[0])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("server_id"), idParts[1])...)
+	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("update_schedule_id"), intId)...)
+	tflog.Info(ctx, "Server update schedule state imported.")
+}
+
+func mapFields(schedule *serverupdate.UpdateSchedule, model *Model) error {
+	if schedule == nil {
+		return fmt.Errorf("response input is nil")
+	}
+	if model == nil {
+		return fmt.Errorf("model input is nil")
+	}
+	if schedule.Id == nil {
+		return fmt.Errorf("response id is nil")
+	}
+
+	model.UpdateScheduleId = types.Int64PointerValue(schedule.Id)
+	idParts := []string{
+		model.ProjectId.ValueString(),
+		model.ServerId.ValueString(),
+		strconv.FormatInt(model.UpdateScheduleId.ValueInt64(), 10),
+	}
+	model.ID = types.StringValue(
+		strings.Join(idParts, core.Separator),
+	)
+	model.Name = types.StringPointerValue(schedule.Name)
+	model.Rrule = types.StringPointerValue(schedule.Rrule)
+	model.Enabled = types.BoolPointerValue(schedule.Enabled)
+	model.MaintenanceWindow = types.Int64PointerValue(schedule.MaintenanceWindow)
+	return nil
+}
+
+// If already enabled, just continues
+func enableUpdatesService(ctx context.Context, model *Model, client *serverupdate.APIClient) error {
+	projectId := model.ProjectId.ValueString()
+	serverId := model.ServerId.ValueString()
+	enableServicePayload := serverupdate.EnableServicePayload{}
+
+	tflog.Debug(ctx, "Enabling server update service")
+	err := client.EnableService(ctx, projectId, serverId).EnableServicePayload(enableServicePayload).Execute()
+	if err != nil {
+		if strings.Contains(err.Error(), "Tried to activate already active service") {
+			tflog.Debug(ctx, "Service for server update already enabled")
+			return nil
+		}
+		return fmt.Errorf("enable server update service: %w", err)
+	}
```

Comment:
Is this the only way to do it? Would be nice if the endpoint was idempotent or if it returned a specific status code. Othwrwise we have to depend on the error message

Reply:
Yes, currently it is. Sorry.